File: jewelcart-backend/src/main/java/com/becoder/stockbackend/services/CustomerService.java
package com.becoder.stockbackend.services;

import com.becoder.stockbackend.model.Customer;
import com.becoder.stockbackend.model.Stock;

import java.util.List;

public interface CustomerService {
    List<Customer> getAllCustomers();
    Customer getCustomerById(Long id);
    Customer saveCustomer(Customer customer);
    Customer updateCustomer(Long id, Customer customer);
    void deleteCustomer(Long id);
    List<Customer> searchCustomers(String keyword);
    List<Customer> filterCustomersByDepartment(String department);
    List<Customer> sortCustomersByName();

    Customer assignStocks(Long id, Long stockId, Customer customer);
}

File: jewelcart-frontend/src/service/customer.service.js
import axios from "axios";

const API_URL = "http://jewel-store.railway.internal/api/customers"; // Update API endpoint for customers

class CustomerService {
    getAllCustomers() {
        return axios.get(API_URL);
    }

    getCustomerById(id) {
        return axios.get(`${API_URL}/${id}`);
    }

    createCustomer(customer) {
        return axios.post(API_URL, customer);
    }

    updateCustomer(id, customer) {
        return axios.put(`${API_URL}/${id}`, customer);
    }

    deleteCustomer(id) {
        return axios.delete(`${API_URL}/${id}`);
    }

    assignStocks(id, stockId, customer) {
        return axios.put(`${API_URL}/${id}/stock/${stockId}`, customer);
    }
}

export default new CustomerService();
